docker/dockerfiles/Dockerfile-dev-ubuntu

```diff
@@ -1,10 +1,34 @@
-FROM starrocks/dev-toolchain-ubuntu:main
+FROM ubuntu:22.04 as base
 
+# Install common libraries and tools that are needed for dev environment
+RUN apt-get update -y && \
+    apt-get install --no-install-recommends -y \
+    automake binutils-dev bison byacc ccache flex libiberty-dev libtool maven zip python3 python-is-python3 make cmake gcc g++ default-jdk git patch lld bzip2 \
+    wget unzip curl vim tree net-tools openssh-client && \
+    rm -rf /var/lib/apt/lists/*
+
+ENV JAVA_HOME=/lib/jvm/default-java
 ENV STARROCKS_THIRDPARTY=/opt/starrocks/thirdparty
 ENV STARROCKS_LINKER=lld
 
 WORKDIR /root
 
-# copy installed thirdparty to the image
-COPY thirdparty/installed /opt/starrocks/thirdparty/installed
+FROM base as builder
+COPY . ./starrocks
```

Comment:
Why not git starrocks code by `git clone`?
this need to download starrocks code first before calling this docker file?

Reply:
And you can see ./bin/build-dev-ubuntu.sh, I use that shell script to build this docker image

Reply:
> Why not git starrocks code by git clone?
it assumes the code has been cloned locally and then this command copy the code to the ununtu container to build. 

I this way, you can easily integrate the release of dev-env image with github action flow:
```
jobs:
  release-starrocks:
    runs-on: [self-hosted, saola-linux]
    steps:
      - name: checkout repository
        uses: actions/checkout@v3

      - DOCKER_BUILDKIT=1 docker build  -f docker/dockerfiles/Dockerfile-dev-ubuntu -t sr-dev-env:${RELEASE_VERSION} .
```

Reply:
you can do this by call `./docker/bin/build-dev-utuntu.sh` too

Reply:
yes. You can do it with external script.
This whole idea  is to use a single Dockerfile with just a `docker build` cmd. You don’t need a separate build script, either. so we could have just one place to watch and maintain instead of touch multiple places to get things done.

Docker community recommends using [multi-stage docker build](https://docs.docker.com/build/building/multi-stage/) 
You can check more detailed discussion in this post  

![image](https://user-images.githubusercontent.com/5108773/214997265-52b887d0-edc5-467a-bedc-472c2a85ec65.png)

Reply:
I have added instruction of kicking off the build:
https://github.com/StarRocks/starrocks/pull/16890/files#diff-e27129ed5bdba5985005b98f6cd71dbcc0d7181ce8cf4d46baec9f643326e9e9R4